### PR TITLE
repo-tools: Add backend-lint command

### PR DESCRIPTION
**Pull request by jhaals**
Added a new command to lint backend plugins so that they support and conform with the new backend system. 

Screenshot 2024-08-21 at 13 24 28

### CODE REVIEW

File: packages/repo-tools/cli-report.md [hunk 1]
```diff
@@ -39,6 +40,15 @@ Options:
   -h, --help
 ```
 
+### `backstage-repo-tools backend-lint`
```
Comment (Rugvip) [inline packages/repo-tools/cli-report.md:43]: Kinda wondering if there's a better structure here, something like `backstage-repo-tools lint legacy-backend-exports` or so? Just leaving space for adding some more of these without too much clutter?

File: packages/repo-tools/src/commands/backend-lint/backend-lint.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fs from 'fs-extra';
+import { Project } from 'ts-morph';
+import { resolvePackagePaths } from '../../lib/paths';
+
+const project = new Project({
+  tsConfigFilePath: 'tsconfig.json',
```
Comment (Rugvip) [inline packages/repo-tools/src/commands/backend-lint/backend-lint.ts:21]: Best use `paths.resolveTargetRoot('tsconfig.json')` here, exported alongside `resolvePackagePaths`.

File: packages/repo-tools/src/commands/backend-lint/backend-lint.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+});
+
+export async function lint(paths: string[]) {
+  const pkgs = (await resolvePackagePaths()).filter(pkg => {
```
Comment (Rugvip) [inline packages/repo-tools/src/commands/backend-lint/backend-lint.ts:29]: Seeing as we don't filter at all it's probably cleaner to use `PackageGraph.listTargetPackages()` directly. That avoids the need to load in the `package.json` files again.

**Review by Rugvip — COMMENTED** [review]
Nice! 👍 

Just some stuff 😁

File: packages/repo-tools/cli-report.md (hunk 1 above)
Comment (jhaals) [reply packages/repo-tools/cli-report.md:43]: I like that idea 👍

**Review by Rugvip — APPROVED** [review]
:shipit: 🎉

**Pull request merged**